### ISSUE TITLE
[Verification] How to add header in ProviderResponseBuilder?

### ISSUE THREAD

**Issue by Arknot** [jbwheatley/pact4s#139]
Hi,

I have try to add a header in [ProviderResponseBuilder](https://github.com/jbwheatley/pact4s/blob/20042fe7cae3dbbdc6bc2074b8c23bac269da3b8/shared/src/main/scala/pact4s/provider/ResponseBuilder.scala#L53) with the below code.

        ProviderResponseBuilder(
          responseCode,
          responseBody
        )
          .withHeaders(Map("Content-Type" -> List[String]("application/json;charset=UTF-8")))

However, I got pact verification failed with below error.
`scala.collection.immutable.$colon$colon cannot be cast to java.util.List`

As I try to debug, it seems the type of Header is different where expected response header type is `ArrayList`, but the actual response header type is `::`.

Any idea how to fix this?

**Comment by jbwheatley** [comment]
Ah, this looks like a bug. This: 

https://github.com/jbwheatley/pact4s/blob/2cafb51e1093e034cddca174633c9f622d7591d3/shared/src/main/scala/pact4s/provider/ResponseBuilder.scala#L62

isn't converting the `scala.List` in`headers: Map[String, List[String]]` to java lists.

**Comment by jbwheatley** [comment]
as a local fix before we publish a new version with this fix, you could override the `build` method on

**Comment by Arknot** [comment]
For the code
https://github.com/jbwheatley/pact4s/blob/2cafb51e1093e034cddca174633c9f622d7591d3/shared/src/main/scala/pact4s/provider/ResponseBuilder.scala#L62

As I understand, the root cause is the map was cast to Java map, but the list inside was not.
So, It should work if we change into this. Please correct me if I'm wrong. 
```
 "headers"     -> headers.map{ case (k, v) => k -> v.asJava}.asJava
```

**Comment by jbwheatley** [comment]
released in v0.1.0
